pkg/url/url.go

```diff
@@ -0,0 +1,53 @@
+// package url provides operations on URLs and URL schemes for log forwarding destinations.
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// ParseAbsolute parses an absolute URL - one with non-empty Scheme and Host.
+// Returned errors include the string s.
+func ParseAbsolute(s string) (*URL, error) {
+	if s == "" {
+		return nil, fmt.Errorf("empty")
+	}
+	u, err := url.Parse(s)
+	switch {
+	case err != nil:
+		return nil, err
+	case u.Scheme == "":
+		return nil, fmt.Errorf("no scheme: %v", u)
+	case u.Host == "":
+		return nil, fmt.Errorf("no host: %v", u)
+	}
```

Comment:
What about `case u.Port != 0` ?

Reply:
Could do that but it's crossing from syntactic validation to semantic, which I wasn't intending here.
http://foo:0 is a valid URL even if not a useful one. Actually http://foo:blahblah is a valid URL, since symbolic port names are allowed and the URL spec doesn't define the list of legal values. The actual resolvable port names depends on the local OS, contents of /etc/services etc.